include/wlr/render/color.h

```diff
@@ -0,0 +1,19 @@
+#ifndef _WLR_COLOR_H
+#define _WLR_COLOR_H
+
+struct wlr_color_config {
+    char *icc_profile_path; // not null
```

Comment:
You need a colorspace here as well. The colorspace is as good of a substitute of a profile as an actual profile.
That's why the eternally stuck in WIP color management protocol specifies both.
You need color primaries, transfer curve, and colorspace. And HDR parameters (which I'm not sure the protocol describes, which would be bad).

Reply:
I originally had a colorspace too when I was using libplacebo. However libplacebo just ignores the colorspace when an ICC profile is set. My understanding, and that of the libplacebo developer, is that an ICC profile makes a colorspace redundant.

What would it be used for if we provided it?

Reply:
Because colorspaces are an alternative to ICC profiles. If a colorspace is specified and a client specifies the same colorspace then no conversion should happen. I expect this to be common in the future, especially with HDR. Because you'll only know of a monitor's colorspace rather than get an ICC profile, and if you hope to have HDR on it you'll definitely need something different than bt709.

Reply:
I agree colorspaces are an alternative, but you can do "color profile /usr/share/color/argyll/ref/Rec2020.icm" or some well-known path instead of making an additional config option.

The easier way to tell that the conversion can be skipped would be to check if the CLUT is a 1-to-1 mapping.

I don't know why you wouldn't have an ICC profile for a monitor - usable defaults are provided by colord from the EDID data.

Reply:
EDID data is highly unreliable and inaccurate. I'd trust sRGB over an ICC generated from an EDID. To get a good ICC profile you really need either equipment, or the guarantee of a manufacturer.
And not everyone wants to install argyll at all.

Reply:
You don't need to use the EDID data. You can use a standard profile. I'm not really against hard-coding colorspaces, but it seems to complicate things with no benefit. If you really want to be user friendly, wlroots could include some standard color profiles.